pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -29,6 +41,45 @@ func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context,
 		if ok {
 			dv.platformIdentitiesActionsMap[k] = nil
 			dv.platformIdentities[k] = pwi
+
+			identityResourceId, err := azure.ParseResourceID(pwi.ResourceID)
+			if err != nil {
+				return err
+			}
+
+			// validate federated identity credentials
+			federatedCredentials, err := clusterMsiFederatedIdentityCredentials.List(ctx, identityResourceId.ResourceGroup, identityResourceId.ResourceName, &sdkmsi.FederatedIdentityCredentialsClientListOptions{})
+			if err != nil {
+				return err
+			}
+
+			for _, federatedCredential := range federatedCredentials {
+				if oc.Properties.ProvisioningState == api.ProvisioningStateCreating {
+					return api.NewCloudError(
+						http.StatusBadRequest,
+						api.CloudErrorCodePlatformWorkloadIdentityContainsInvalidFederatedCredential,
+						fmt.Sprintf("properties.platformWorkloadIdentityProfile.platformWorkloadIdentities.%s.resourceId", k),
+						"Unexpected federated credential '%s' found on platform workload identity '%s' used for role '%s'. Please ensure this identity is only used for this cluster and does not have any existing federated identity credentials.",
+						*federatedCredential.Name,
+						pwi.ResourceID,
+						k,
+					)
+				}
+
+				if len(federatedCredential.Properties.Audiences) != 1 ||
+					*federatedCredential.Properties.Audiences[0] != expectedAudience ||
+					*federatedCredential.Properties.Issuer != string(*oc.Properties.ClusterProfile.OIDCIssuer) {
+					return api.NewCloudError(
+						http.StatusBadRequest,
+						api.CloudErrorCodePlatformWorkloadIdentityContainsInvalidFederatedCredential,
+						fmt.Sprintf("properties.platformWorkloadIdentityProfile.platformWorkloadIdentities.%s.resourceId", k),
```

Comment:
Can we also add role/issuer to the error message here to be clear of all the details used to validate?

Reply:
`k` is the role (OperatorName) here. We can include the issuer on the error message, and I think it's worth having these error messages reviewed in detail. I'll bring this up in our next sync meeting.

Reply:
Actually, given the issues with nil pointer dereferences above, I'm in favor of not including the issuer. It could be nil and conditionally handling the nil pointer dereference would introduce too much complexity into the error generation process. We return the full resource ID of the identity and the name of the offending FIC here, which I think should be sufficient for customers to identify what's wrong.